transcoder/src/keyframes.go

```diff
@@ -164,10 +172,46 @@ func getKeyframes(path string, kf *Keyframe) error {
 			ret = ret[:0]
 		}
 	}
+	if len(ret) == 0 {
+		duration, err := getFileDuration(path)
+		if err != nil {
+			return err
+		}
+		ret = make([]float64, 0, int((duration/2)+1))
+		for segmentTime := float64(0); segmentTime < duration; segmentTime += 2 {
+			ret = append(ret, segmentTime)
+		}
+	}
 	kf.add(ret)
 	if done == 0 {
 		kf.info.ready.Done()
 	}
 	kf.IsDone = true
 	return nil
 }
+
+func getFileDuration(path string) (float64, error) {
```

Comment:
You should use GetInfo from the `info.go` file. It has a cache and is already called before accessing keyframes.

Reply:
You should delete the now unused `getFileDuration`